Userland/Libraries/LibJS/Runtime/RegExpPrototype.cpp

```diff
@@ -281,4 +282,148 @@ JS_DEFINE_NATIVE_FUNCTION(RegExpPrototype::symbol_match)
     return vm.call(*exec, rx, js_string(vm, s));
 }
 
+JS_DEFINE_NATIVE_FUNCTION(RegExpPrototype::symbol_replace)
+{
+    auto string_value = vm.argument(0);
+    auto replace_value = vm.argument(1);
+
+    // https://tc39.es/ecma262/#sec-regexp.prototype-@@replace
+    auto rx = regexp_object_from(vm, global_object);
+    if (!rx)
+        return {};
+    auto string = string_value.to_string(global_object);
+    if (vm.exception())
+        return {};
+
+    auto global_value = rx->get(vm.names.global).value_or(js_undefined());
+    if (vm.exception())
+        return {};
+
+    bool global = global_value.to_boolean();
+    if (global)
+        rx->regex().start_offset = 0;
+
+    // FIXME: Implement and use RegExpExec - https://tc39.es/ecma262/#sec-regexpexec
+    auto* exec = get_method(global_object, rx, vm.names.exec);
+    if (!exec)
+        return {};
+
+    Vector<Value> results;
+
+    while (true) {
+        auto result = vm.call(*exec, rx, string_value);
+        if (result.is_null()) {
+            break;
+        }
+
+        results.append(move(result));
+        if (!global) {
+            break;
+        }
+
+        auto* match_object = results.last().to_object(global_object);
+        if (!match_object)
+            return {};
+
+        String match_str = match_object->get(0).to_string(global_object);
+        if (vm.exception())
+            return {};
+        if (match_str.is_empty()) {
+            // FIXME: Implement AdvanceStringIndex to take Unicode code points into account - https://tc39.es/ecma262/#sec-advancestringindex
+            //        Once implemented, step (8a) of the @@replace algorithm must also be implemented.
+            rx->regex().start_offset += 1;
+        }
+    }
+
+    String accumulated_result;
+    size_t next_source_position = 0;
+
+    for (auto& result : results) {
+        auto* result_object = result.to_object(global_object);
+        if (!result_object)
+            return {};
+
+        size_t result_length = length_of_array_like(global_object, *result_object);
+        size_t n_captures = result_length == 0 ? 0 : result_length - 1;
+
+        auto matched = result_object->get("0");
+        if (vm.exception())
+            return {};
+
+        auto position_value = result_object->get(vm.names.index);
+        if (vm.exception() || position_value.is_empty())
+            return {};
```

Comment:
I think this would be

```suggestion
        auto position_value = result_object->get(vm.names.index).value_or(js_undefined());
        if (vm.exception())
            return {};
```

undefined `to_integer_or_infinity()`'d results in `0`, but we can continue with that